study-frontend/src/Components/Header.tsx

```diff
@@ -1,27 +1,186 @@
+import { CgProfile } from "react-icons/cg";
+import { BsFilterLeft } from "react-icons/bs";
+import { useNavigate } from 'react-router-dom';
+import { routes } from '../constants';
+import { RiArrowDropDownLine } from "react-icons/ri";
+import { RiArrowDropUpLine } from "react-icons/ri";
+import FilterDropDown from './FilterDropDown';
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface MenuPage{
+  name: string,
+  rout: string, 
+  style: string,
+ 
+}
 
 type Props = {
-  handleFaBar: ()=>void
+  handleFaBar: ()=>void,
+  handleFilterDropDown: ()=>void
+  handleProfileBar: () => void
+  filterDropDown : boolean
 }
 
-const Header = ({handleFaBar}: Props) => {
+const Header = ({handleFaBar, handleProfileBar, handleFilterDropDown, filterDropDown}: Props) => {
+  const [viewportWidth, setViewportWidth] = useState(window.innerWidth);
+  const [searchValue, setSearchValue] = useState("");
+  const navigator = useNavigate();
+  React.useEffect(() => {
+
+    window.addEventListener("resize", () => setViewportWidth(window.innerWidth));
+
+
+  }, []);
+
+  const pages : MenuPage[] = [
+    {
+      name: "Home Page",
+      rout: "home",
+      style: "NavBarButtons",
+
+    },
+    {
+      name: "Messeges",
+      rout: "searchPage",
+      style: "NavBarButtons",
+
+    },
+    {
+      name: "Favorites",
+      rout: "favorites",
+      style: "NavBarButtons",
+
+    },
+    {
+      name: "History",
+      rout: "history",
+      style: "NavBarButtons",
+
+    },
+
+  ];
+
+
+  
+
   return (
     <div className='closedSideBar'>
-        
+      {viewportWidth < 780 ? (
+        <React.Fragment>
           <div className='MenuIcon'>
-              <FaBars onClick={handleFaBar}/>
+            <HiMenuAlt2   onClick={handleFaBar} />
           </div>
-          
           <div className='closedSideBarText'>
-              Study
+            Study
           </div>
-          {/* <input type="text" placeholder="Search" className="input input-bordered w-24 md:w-auto" /> */}
-    </div>
-        
+          <button id='SearchIconBox' onClick={()=>{
+             navigator(`/search/`, {state :{searchInfo : "", tagName : ""}} )
+          }}>
+
+            <CiSearch id='SearchIcon' />
+
+          </button >
+        </React.Fragment>
+      ) : (
+        <React.Fragment>
+          {viewportWidth < 1250 ? 
+          (
+            <React.Fragment>
+              <div className='MenuIcon'>
+                <HiMenuAlt2   onClick={handleFaBar} />
+              </div>
+              <div className='closedSideBarText'>
+                Study
+              </div>
+
+              
+              
+              <div className='SearchBarField' style={{height: "100%"}}>
+                <input type="text" placeholder="Search" className="SearchBar" value={searchValue} onChange={(e) =>{          
+                  setSearchValue(e.target.value);
+                  console.log("hi")
+                  if(e.target.value!== "")
```

Comment:
You are setting the search before you are checking value.
Make sure that your logic should be I do checks if check pass do action to be more readable